upgrades/upgrade_testing_framework/core/logging_wrangler.py

```diff
@@ -0,0 +1,54 @@
+import coloredlogs
+from datetime import datetime
+import logging
+import os
+
+from upgrade_testing_framework.core.workspace_wrangler import WorkspaceWrangler
+
+class FrameworkLoggingAdapter(logging.LoggerAdapter):
+    # Print the name of the currently-executing step, then the log message
+    def process(self, msg, kwargs):
+        return "[{}] {}".format(self.extra['step'], msg), kwargs
+
+class FrameworkLoggingFormatter(logging.Formatter):
+     def formatTime(self, record, datefmt=None):
+        return datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S.%f')
+
+LINE_SEP = '\u2063' # Invisible Unicode character.  Makes figuring out the beginning/end of a log entry much
+            # easier, given they can contain new line characters themselves.
+
+class LoggingWrangler:
```

Comment:
Is this wrangler something we should be using for all our logging operations? Without this I imagine our DEBUG logs for those operations wouldn't appear in this timestamped logfile

Reply:
Yeah, it should be used everywhere, but you only need to set it up once in a given Python application.  Check the Python library docs on logging to get a better feel for how it works.  It's a bit complicated.